src/Illuminate/Auth/Passwords/PasswordBroker.php

```diff
@@ -112,7 +112,7 @@ public function emailResetLink(CanResetPasswordContract $user, $token, Closure $
         return $this->mailer->send($view, compact('token', 'user'), function ($m) use ($user, $token, $callback) {
             $m->to($user->getEmailForPasswordReset());
 
-            if (! is_null($callback)) {
+            if ($callback instanceof Closure) {
                 call_user_func($callback, $m, $user, $token);
```

Comment:
@crynobone How is that? The callback already has a hint of type `Closure`, and a default of `null`. Instead of checking the negation of `is_null()` I do the check using `instanceof`. The method signature does not change.